exporter/collector/logs.go

```diff
@@ -376,16 +376,20 @@ func (l logMapper) logToSplitEntries(
 	logName string,
 	projectID string,
 ) ([]logging.Entry, error) {
+	// make a copy in case we mutate the record
```

Comment:
Is this necessary? I don't see the usages of `log` variable in this method indicating that we don't care about the original object and its not a pointer, so I am assuming its pass by value.

Reply:
the [underlying pdata](https://github.com/open-telemetry/opentelemetry-collector/blob/38615b2ad813a277e4a4fe0b391f3cc43db975b2/pdata/plog/generated_logrecord.go#L24) is a pointer, so collector components aren't supposed to mutate data which could be shared (like across multiple exporters) https://github.com/open-telemetry/opentelemetry-collector/tree/89e13a1d94c230cd1552d5a73e6730dace710921/exporter#data-ownership